app/queries/cohort_analytics_query.rb

```diff
@@ -5,17 +5,17 @@ def initialize(patients, period = :month)
     @patients = patients
   end
 
-  def patient_counts_by_period(period, prev_periods)
+  def patient_counts_by_period(period, prev_periods, from_date: Date.today)
     results = {}
 
     (0..(prev_periods)).each do |periods_back|
       if period == :month
-        report_start = (Time.current - periods_back.months).beginning_of_month
-        report_end   = report_start.end_of_month
+        report_start = (from_date - (1 + periods_back).months).beginning_of_month
```

Comment:
If we aren't skipping the current quarter, I think we shouldn't skip the current month either. We will have to actually wait longer for the current quarter to have complete data, than for a month.

Reply:
This is a design decision we made a while ago. I agree that the inconsistency is potentially confusing and makes the code more complex.

@dburka What do you think of only showing the most recently completed quarter or month, not showing the current quarter or month?

Showing the current quarter allows people to see how they're doing right now, as opposed to waiting 3 months. Of course, "didn't attend" for the current quarter includes patients that just haven't had time to show up yet, inflating the numbers a bit.

Showing the "current month" is tricky, as monthly cohort spans 2 months (e.g. followed up between Feb and March). This code skips the current month and only shows the last completed monthly cohort (from last month).